src/uct/base/uct_mem.c

```diff
@@ -212,16 +231,19 @@ ucs_status_t uct_mem_alloc(void *addr, size_t min_length, unsigned flags,
                 goto allocated_without_md;
             }
 
-            ucs_trace("failed to mmap %zu bytes: %s", min_length,
+            ucs_trace("failed to mmap %zu bytes: %s", param->length,
                       ucs_status_string(status));
             break;
 
         case UCT_ALLOC_METHOD_HUGE:
+            if (mem_type != UCS_MEMORY_TYPE_HOST) {
+                break;
+            }
 #ifdef SHM_HUGETLB
             /* Allocate huge pages */
-            alloc_length = min_length;
-            address = (flags & UCT_MD_MEM_FLAG_FIXED) ? addr : NULL;
-            status = ucs_sysv_alloc(&alloc_length, min_length * 2, &address,
+            address = (flags & UCT_MD_MEM_FLAG_FIXED) ?
+                      param->address : NULL;
```

Comment:
check that address is set

Reply:
This is already checked in uct_mem_alloc_check_params 
https://github.com/openucx/ucx/blob/f4d054c1df22a9072212db9df398cebf243dc35c/src/uct/base/uct_md.c#L398

Reply:
ok